integration-tests/src/test/java/org/jenkinsci/bytecode/CompatibilityTest.java

```diff
@@ -83,9 +84,13 @@ private AntClassLoader createClassLoader(String v) throws IOException {
                 return super.defineClassFromData(container, rewritten, classname);
             }
         };
-
-        cl.addPathComponent(new File("target/test-classes/"+ v));
-        cl.addPathComponent(new File("target/test-classes/client"));
+        FileFilter jarFilter = file -> file.getName().endsWith(".jar");
+        for (File jarFile : new File("../test-" + v + "/target").listFiles(jarFilter)) {
+            cl.addPathComponent(jarFile);
```

Comment:
The `AntClassLoader#addPathComponent` Javadoc makes me think that adding a directory should work, but for whatever reason it does not. I will file a followup ticket to investigate and fix the documentation and/or code.